File: 1.0/src/c/cz/mg/vulkan/PFN_vkAllocateCommandBuffers.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkAllocateCommandBuffers_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pAllocateInfo, jlong pCommandBuffers, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkAllocateCommandBuffers f = (PFN_vkAllocateCommandBuffers)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkCommandBufferAllocateInfo*)jniLongToPointer(pAllocateInfo)),
        ((VkCommandBuffer*)jniLongToPointer(pCommandBuffers))
    );
}


jint Java_cz_mg_vulkan_PFNvkAllocateCommandBuffers_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pAllocateInfo, jlong pCommandBuffers){
    (void)env;
    (void)jc;
    PFN_vkAllocateCommandBuffers f = (PFN_vkAllocateCommandBuffers)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkCommandBufferAllocateInfo*)jniLongToPointer(pAllocateInfo)),
        ((VkCommandBuffer*)jniLongToPointer(pCommandBuffers))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkAllocateCommandBuffers.java
package cz.mg.vulkan;

public class PFNvkAllocateCommandBuffers extends VkFunctionPointer {
    public PFNvkAllocateCommandBuffers() {
    }

    protected PFNvkAllocateCommandBuffers(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkAllocateCommandBuffers(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkAllocateCommandBuffers(long value) {
        setValue(value);
    }

    public PFNvkAllocateCommandBuffers(VkInstance instance) {
        super(instance, new VkString("vkAllocateCommandBuffers"));
    }

    public void call(VkDevice device, VkCommandBufferAllocateInfo pAllocateInfo, VkCommandBuffer pCommandBuffers, VkResult rval){
        callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pAllocateInfo != null ? pAllocateInfo.getVkAddress() : VkPointer.NULL, pCommandBuffers != null ? pCommandBuffers.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void callNative(long vkaddress, long device, long pAllocateInfo, long pCommandBuffers, long rval);


    public int call(VkDevice device, VkCommandBufferAllocateInfo pAllocateInfo, VkCommandBuffer pCommandBuffers){
        return callSimplifiedNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), pAllocateInfo != null ? pAllocateInfo.getVkAddress() : VkPointer.NULL, pCommandBuffers != null ? pCommandBuffers.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callSimplifiedNative(long vkaddress, long device, long pAllocateInfo, long pCommandBuffers);

}
